featuretools/primitives/standard/rolling_transform_primitive.py

```diff
@@ -0,0 +1,335 @@
+import numpy as np
+import pandas as pd
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Datetime, Double
+
+from featuretools.primitives.base.transform_primitive_base import (
+    TransformPrimitive
+)
+from featuretools.primitives.utils import _roll_series_with_gap
+
+
+class RollingMax(TransformPrimitive):
+    """Determines the maximum of entries over a given window.
+
+    Description:
+        Given a list of numbers and a corresponding list of
+        datetimes, return a rolling maximum of the numeric values,
+        starting at the row `gap` rows away from the current row and looking backward
+        over the specified window (by `window_length` and `gap`).
+
+        Input datetimes should be monotonic.
+
+    Args:
+        window_length (int): The number of rows to be included in each frame. For data
+            with a uniform sampling frequency, for example of one day, the window_length will
+            correspond to a period of time, in this case, 7 days for a window_length of 7.
+        gap (int, optional): The number of rows prior to each instance to be skipped before
```

Comment:
How's this sound

> The number of rows backward from the target instance to be skipped before the window of usable data begins.  Defaults to 0, which will include the target instance in the window

If `include_cutoff_time` is False does gap=0 function the same as gap=1 would normally?

Reply:
I like your suggestion. The one thing I might change is just completely getting rid of the word "skipped" based off of @freddyaboulton's comment above. How about

> The number of rows backward from the target instance before the window of usable data begins. Defaults to 0, which will include the target instance in the window

I think just talking about "n rows back" better encapsulates the idea that one row back is immediately above and zero rows back is just that instance. Instead of also including the idea of "n rows skipped" which would lead me to think there would be `n` entire rows that are ignored before starting the window.

Reply:
To your second point: unless I'm missing something, setting a cutoff time and having `include_cutoff_time=False` shouldn't have an impact on the values included in any window. 

In the example below where we have primitives set with gap=0 and gap=1 and it just impacts the final row.
![image](https://user-images.githubusercontent.com/64278226/140983138-a63e2144-63bc-4733-a3f0-d3e3690de0b2.png)

If I set cutoff_times for each row to be its own time index value (and now we have to have include_cutoff_time=True or else we'll have no data even though windows technically would be before the cutoff time--may need to be a separate issue if we wanted to handle that case specifically), it has no impact at all. (If we also add `uses_full_dataframe = True` to the primitive because of that separate bug)

![image](https://user-images.githubusercontent.com/64278226/140986052-8b6c49f7-d373-49d0-a541-dac84e201e50.png)

Reply:
Thanks for the explanation, I think I was confusing using the cutoff time to create a window with using the target instance for a window

Reply:
Ahh fair. Anything I can add to the docstrings to help clarify what happens in these primitives?

Reply:
I think it's pretty clear

Reply:
We could put this information in our documentation. Add it as a section or a new page on Time Series. Use it as a chance to explain rolling primitives

Reply:
Have an issue to do that (#1758 - could also just be adding an entirely new Time series page). Was imagining it could happen once we have a lot of the time series stuff in, but we could always start it now and update it as we go along

Reply:
@tamargrey your call. happy with either.

Reply:
Gonna leave the documentation to later. Should be a bit easier to create a more cohesive narrative when everything is implemented.